### PR TITLE
Translate to [pt-br] Chapter 12 (Tips and Tricks) and Chapter 13 (Moving Forward).

### PR DESCRIPTION
**Pull request by besser**
Translate to [pt-br] Chapter 12 (Tips and Tricks) and Chapter 13 (Moving Forward).

### CODE REVIEW

File: en/tips_and_tricks/README.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ func MyHandler(database *sql.DB) http.Handler {
 ## Using `gorilla/context` for request-specific data
 It is pretty often that we need to store and retrieve data that is specific to
 the current HTTP request. Use `gorilla/context` to map values and retrieve them
-later. It contains a global mutex on a map of request objects.
+later. It contains a global mutex on a map of requested objects.
```
Comment (carlisia) [inline en/tips_and_tricks/README.md]: Aqui nao deveria mudar request para requested, request objects estah correcto (request objects, response objects, as in types of objects)

File: pt-br/tips_and_tricks/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Tips and Tricks
+
+## Envolva um http.HandlerFunc com closure (contexto léxico)
+As vezes você quer passar dados na inicialização de um http.HandlerFunc. Isto pode ser facilmente feito criando um `closure` de `http.HandlerFunc`:
+
+``` go
+func MyHandler(database *sql.DB) http.Handler {
+  return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+    // you now have access to the *sql.DB here
```
Comment (carlisia) [inline pt-br/tips_and_tricks/README.md]: deveria traduzir esse comentario?

File: pt-br/tips_and_tricks/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+```
+
+## Usando `gorilla/context` para requisição específica de dados
+Com frequência precisamos armazenar e recuperar dados que são específicos da requisição HTTP atual. Use o `gorilla/context` para mapear valores e recuperá-los mais tarde. Ele contém um `mutex` (mutual exclusion, ou exclusão mútua na tradução livre) global do mapa de objetos requisitados.
+
+``` go
+func MyHandler(w http.ResponseWriter, r *http.Request) {
+    val := context.Get(r, "myKey")
+
+    // returns ("bar", true)
```
Comment (carlisia) [inline pt-br/tips_and_tricks/README.md]: deveria traduzir o comentario

**Comment by carlisia** [comment]
Fora os comentarios, :+1:

**Comment by besser** [comment]
Carlisia, fiz as alterações que vc solicitou!

**Pull request merged**